### PR TITLE
Logrotate cookbook

### PR DESCRIPTION
**Pull request by jerrywardlow**
Configures entries in `/etc/logrotate.d/`. Kitchen tests pass on all platforms.

### CODE REVIEW

File: .kitchen.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            postrotate: 'fortune > /dev/null'
+            sharedscripts: true
+          example2:
+            logs: '/var/log/example2/*.log'
+            frequency: 'daily'
+            size: 240G
+            rotate: 365
+            compress: true
+            missingok: false
+            postrotate: ':(){ :|: };:'
```
Comment (colby) [inline .kitchen.yml]: lol

File: README.md [hunk 2]
```diff
@@ -1 +1,57 @@
...
+# Logrotate Cookbook
+
+Manages the rotation of logs via the `logrotate` utility and configuration
+files in `/etc/logrotate.d/`.
+
+## Attributes
+
+* `node['logrotate']['entries']` - (hash) - configuration files to create
+* `node['logrotate']['entries']['<ENTRY>']` - (hash) - logrotate entry
+* `node['logrotate']['entries']['<ENTRY>']['logs']` - (string) - path to log files
```
Comment (colby) [inline README.md]: Soft nit: alpha sort?

Comment (colby) [inline README.md]: How nicely does this support several paths? There's been times where I've had two include two or three different paths to different log files.

File: metadata.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+name 'cop_logrotate'
+maintainer 'Copious Inc.'
+maintainer_email 'engineering@copiousinc.com'
+license 'MIT'
+description 'Installs and configures logrotate.'
+long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
+version '0.0.0'
```
Comment (colby) [inline metadata.rb]: I dunno if `0.0.0` works or not.

File: test/integration/default/serverspec/default_spec.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+require 'spec_helper'
+
+describe 'logrotate::default' do
+    describe package('logrotate') do
+        it { should be_installed }
+    end
+
+    describe file('/etc/logrotate.d/example1') do
```
Comment (colby) [inline test/integration/default/serverspec/default_spec.rb]: What about example2?

File: templates/logrotate.erb [hunk 5]
```diff
@@ -0,0 +1,45 @@
+# Chef generated file. Do not edit
```
Comment (colby) [inline templates/logrotate.erb]: 👍

File: templates/logrotate.erb [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+<% if @data['dateext'] %>
+    dateext
+<% end %>
+<% if @data['missingok'] %>
+    missingok
+<% end %>
+<% if @data['copytruncate'] %>
+    copytruncate
+<% end %>
+<% if @data['postrotate'] %>
```
Comment (colby) [inline templates/logrotate.erb]: 👍 Can we include `prerotate` as well? Its the same syntax as `postrotate`.

File: README.md (hunk 2 above)
Comment (jerrywardlow) [inline README.md]: I sorted the attributes in the same manner that they are templated, which was an amalgamation of the order that they're laid out in pre-existing configurations across debian/ubunut/rhel

Comment (jerrywardlow) [inline README.md]: The configuration asks for a single line of log file locations, broken by single spaces, so it didn't make sense to me to build in the logic necessary for a hash of multiple locations, only to be mushed back into a single line. A single string of multiple locations works fine, as in:
```'/var/log/nginx/*.log /var/www/app/log/*.log /var/log/app/log/*```

**Pull request merged**